src/app/core/params.ts

```diff
@@ -0,0 +1,215 @@
+import Parameter from "./parameter";
+/**
+ * Params class
+ */
+export default class Params {
+    private unique: boolean;
+    private heritable: boolean;
+    parameters: any;
+    
+
+    /**
+     * Default constructor create a Params object.
+     * @param {} values Values of the params
+     * @param {Boolean} unique Boolean if it's unique
+     * @param {} heritable Boolean if it's heritable
+     * @param {} rawparameters ?
+     */
+    constructor(values: any, unique: boolean, heritable: boolean) { 
+        //rp must be optional for function further down. If no rp provided and values == null, what happens?
+        this.unique = unique;
+        this.heritable = heritable;
+    }
+    /**
+     * Updates parameter value.
+     * @param {String} key Identifier of the parameter
+     * @param {*} value New value of the parameter
+     * @memberof Params
+     * @returns {void}
+     */
+    updateParameter(key: string, value: any) {
+        if (this.parameters.hasOwnProperty(key)) {
+            this.parameters[key].updateValue(value);
+        } else {
+            throw new Error(key + "parameter does not exist in Params object");
+        }
+    }
+    /**
+     * Checks if the object has certain parameter.
+     * @param {String} key The key is use to identify the parameter 
+     * @memberof Params
+     * @returns {void}
+     */
+    __ensureHasKey(key: string) {
+        if (!this.parameters.hasOwnProperty(key)) throw new Error(key + " parameter not found in Params object.");
+    }
+    /**
+     * Gets the value of the selected parameter.
+     * @param {String} key The key is needed to identify the parameter
+     * @returns Returns parameter value
+     * @memberof Params
+     */
+    getValue(key: string) {
+        this.__ensureHasKey(key);
+        return this.parameters[key].getValue();
+    }
+    /**
+     * Gets the paramter.
+     * @param {String} key The key is needed to search and identify the parameter
+     * @returns Returns parameter
+     * @memberof Params
+     */
+    getParameter(key: string) {
+        this.__ensureHasKey(key);
+        return this.parameters[key];
+    }
+    /**
+     * Checks if param object has unique key.
+     * @param {String} key Key to identify the param
+     * @returns {boolean} 
+     * @memberof Params
+     */
+    isUnique(key: string) {
+        return this.unique.hasOwnProperty(key);
```

Comment:
This need to change to `Object.hasOwnProperty(this.unique, key)`

Reply:
When I attempt to implement this, I get an error stating only one argument is expected. I also cannot find documentation for more than one argument being used with hasOwnProperty.